cmd/limactl/main.go

```diff
@@ -77,7 +77,7 @@ func newApp() *cobra.Command {
 			logrus.SetLevel(logrus.DebugLevel)
 		}
 
-		if osutil.IsBeingRosettaTranslated() {
+		if osutil.IsBeingRosettaTranslated() && cmd.Parent().Name() != "completion" && cmd.Name() != "generate-doc" && cmd.Name() != "validate" {
```

Comment:
Maybe just check `name == "start"` ?

Reply:
I don't know; does it make sense to let the user run `create` but then fail on `start`? Same with most other commands; they are not really useful unless you can actually start an instance.

This PR is mostly just an affordance for developers/packagers to do cross-architecture builds, and not really useful for general users. So catching the arch mis-match for all other commands as soon as possible seems still desirable.

Reply:
If the change should be merged as-is, then I would like to see a comment above that explains why these 3 commands have been explicitly exempted from the Rosetta check (basically what the PR description says).

Reply:
@AkihiroSuda I started looking through commands that are there and found some other that also don't depend on the architecture, but I agree with @jandubois that allowing only `start` to throw an error would worsen user experience.
I've added the comment as @jandubois suggested.